src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,135 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Parser {
+    public static void parse(Storage storage, TaskList tasks, Ui ui) {
+        boolean isRunning = true;
+
+        while (isRunning) {
+            try {
+                String input = ui.readCommand();
+                String[] inputSplit = input.split(" ", 2);
+
+                Command cmd = Command.getCommand(inputSplit[0]);
+
+                switch (cmd) {
```

Comment:
Very organized and complete switch cases!
Not sure about having the main loop inside the parser class, but it works.